File: SistemaBodegaJaime/BodegaJaimeBackend/src/main/java/com/proyectosdejava/service/MarcaService.java
package com.proyectosdejava.service;

import com.proyectosdejava.repository.MarcaRepository;
import jakarta.transaction.Transactional;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Service
public class MarcaService {

    private final MarcaRepository repository;

    public MarcaService(MarcaRepository repository) {
        this.repository = repository;
    }

    public List<Object[]> buscarActivas(String nombre, String empresa) {
        return repository.buscarActivas(nombre, empresa);
    }

    public List<Object[]> buscarInactivas(String nombre, String empresa) {
        return repository.buscarInactivas(nombre, empresa);
    }

    @Transactional
    public Map<String, String> crear(String nombre, String empresa) {
        return buildRespuesta(repository.crear(nombre, empresa));
    }

    @Transactional
    public Map<String, String> editar(Integer id, String nombre, String empresa) {
        return buildRespuesta(repository.editar(id, nombre, empresa));
    }

    @Transactional
    public Map<String, String> cambiarEstado(Integer id, Integer estado) {
        return buildRespuesta(repository.cambiarEstado(id, estado));
    }

    private Map<String, String> buildRespuesta(String resultado) {
        Map<String, String> respuesta = new HashMap<>();
        respuesta.put("mensaje", resultado);
        return respuesta;
    }
}

File: SistemaBodegaJaime/BodegaJaimeFrontend/src/services/marcaService.js
import api from './api'

const marcaService = {

    buscarActivas: (nombre = null, empresa = null) =>
        api.get('/marca/listar-activas', { params: { nombre, empresa } }),

    buscarInactivas: (nombre = null, empresa = null) =>
        api.get('/marca/listar-inactivas', { params: { nombre, empresa } }),

    crear: (datos) =>
        api.post('/marca', datos),

    editar: (id, datos) =>
        api.put(`/marca/${id}`, datos),

    cambiarEstado: (id, valor) =>
        api.patch(`/marca/${id}/estado`, null, { params: { valor } }),
}

export default marcaService